test/src/code_field/code_controller_folding_editing_test.dart

```diff
@@ -146,9 +151,297 @@ int n;
+            const example = 'a\na\na\nmethod1(){\n  if (true) return;\n}\n';
+
+            final controller = await pumpController(wt, example);
+            controller.foldAt(3);
+            await wt.selectFromHome(0, offset: 6);
+
+            const expected = TextEditingValue(
+              text: 'method1(){\n',
+              selection: TextSelection.collapsed(offset: 0),
+            );
+
+            controller.value = controller.value.replacedSelection('');
+            expect(controller.value, expected);
+          });
+        });
+
+        group('Python', () {
+          testWidgets(
+              'Indent block is opened '
+              'if the next line is being edited to be a part of this block',
+              (wt) async {
+            const example = 'a:\n  aaaa\n';
+            final controller = await pumpController(
+              wt,
+              example,
+              language: python,
+            );
+            controller.foldAt(0);
+
+            expect(controller.value.text, 'a:\n');
+            //                                 \ selection
+            await wt.selectFromHome(3);
+            controller.value = controller.value.replacedSelection('  aaaa');
+
+            const expectedText = 'a:\n  aaaa\n  aaaa';
+
+            expect(controller.value.text, expectedText);
+            expect(controller.code.foldedBlocks.length, 0);
+          });
+
+          testWidgets(
+              'Indent block is not opened if the next line is edited '
+              'but it is not part of the block yet', (wt) async {
+            const example = 'a:\n  aaaa\n';
+            final controller = await pumpController(
+              wt,
+              example,
+              language: python,
+            );
+            controller.foldAt(0);
+
+            expect(controller.value.text, 'a:\n');
+            //                                 \ selection
+            await wt.selectFromHome(3);
+            controller.value = controller.value.replacedSelection('aaaa');
+
+            const expectedText = 'a:\naaaa';
+
+            expect(controller.value.text, expectedText);
+            expect(controller.code.foldedBlocks.length, 1);
+
+            controller.unfoldAt(0);
+
+            expect(controller.value.text, 'a:\n  aaaa\naaaa');
+            expect(controller.code.foldedBlocks.length, 0);
+          });
+
+          testWidgets(
+              'Indent block is not opened '
+              'if the newline char was inserted to the next line ', (wt) async {
+            const example = 'a:\n  aaaa\n\n';
+            final controller = await pumpController(
+              wt,
+              example,
+              language: python,
+            );
+
+            controller.foldAt(0);
+
+            expect(controller.value.text, 'a:\n\n');
+            //                                 \ selection
+            await wt.selectFromHome(3);
+
+            controller.value = controller.value.replacedSelection('\na');
```

Comment:
```suggestion
            controller.value = controller.value.replacedSelection('\n');
```
It a timer issue arises, debug it.